### PR TITLE
Matching logic

### PR DESCRIPTION
**Pull request by stmitt**

### CODE REVIEW

File: Sources/DP3TSDK/DP3TTracingState.swift [hunk 1]
```diff
@@ -12,17 +12,17 @@ public enum InfectionStatus {
...
     case healthy
     /// The user was in contact with a person that was flagged as infected
-    case exposed(days: [MatchedContact])
+    case exposed(days: [ExposureDay])
     /// The user is infected and has signaled it himself
     case infected
 
     static func getInfectionState(with database: DP3TDatabase) -> InfectionStatus {
-        let numberOfMatchedContacts = try? database.matchedContactsStorage.count()
+        let numberOfMatchedContacts = try? database.exposureDaysStorage.count()
```
Comment (simonroesch) [inline Sources/DP3TSDK/DP3TTracingState.swift:20]: we should probably rename or remove this, right?

File: Sources/DP3TSDK/DP3TMatching.swift [hunk 2]
```diff
@@ -50,16 +50,46 @@ class DP3TMatcher: DP3TMatcherProtocol {
...
+        // we can return here if we didn't find any new matching contacts
+        guard matchingContacts.isEmpty == false else {
+            return
+        }
+
+        guard let knownCaseId = try? database.knownCasesStorage.getId(for: knownCase.key) else {
+            fatalError("Known case has to be in database at this point")
+        }
+
+        /// Store store all matching links in database
```
Comment (joseph-elmallah) [inline Sources/DP3TSDK/DP3TMatching.swift:62]: double store in documentation

File: Sources/DP3TSDK/DP3TTracingState.swift (hunk 1 above)
Comment (stmitt) [reply Sources/DP3TSDK/DP3TTracingState.swift]: Yes!

File: Sources/DP3TSDK/DP3TMatching.swift [hunk 3]
```diff
@@ -50,16 +50,46 @@ class DP3TMatcher: DP3TMatcherProtocol {
...
+        let matchedDays = groups.compactMap { (day, windowCount) -> ExposureDay? in
+            guard windowCount > ContactFactory.numberOfWindowsForExposure else { return nil }
+            return ExposureDay(identifier: nil,
+                               exposedDate: day,
+                               reportDate: Date())
+        }
+
+        let daysBefore = try database.exposureDaysStorage.getExposureDays()
+
+        /// Safe the matchedDays
```
Comment (joseph-elmallah) [inline Sources/DP3TSDK/DP3TMatching.swift:86]: Typo: Safe -> Save

**Pull request merged**